src/webextension/server.cpp

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "./server.h"
+
+#include <QDebug>
+#include <QHostAddress>
+#include <QJsonObject>
+#include <QTcpSocket>
+
+#include "connection.h"
+
+constexpr int SERVER_PORT = 8754;
+
+namespace WebExtension {
+
+Server::Server(BaseAdapter* adapter) {
+  m_adapter = adapter;
+
+  qInfo() << "Creating the server";
+
+  if (!listen(QHostAddress::LocalHost, SERVER_PORT)) {
+    qCritical() << "Failed to listen on port" << SERVER_PORT;
+    return;
+  }
+
+  connect(this, &Server::newConnection, this, &Server::newConnectionReceived);
+}
+
+Server::~Server() {}
+
+bool Server::isAllowedToConnect(QHostAddress addr) {
+  // `::ffff:127.0.0.1` is the IPv4 localhost address written with the IPv6
+  // notation.
+  return addr == QHostAddress("::ffff:127.0.0.1") ||
```

Comment:
I think I already cmmented this, but I don't see it anymore. Consider making this into a constant.